lib/msf/core/db_manager/host.rb

```diff
@@ -54,13 +54,28 @@ def find_or_create_host(opts)
 
   def add_host_tag(opts)
     wspace = Msf::Util::DBManager.process_opts_workspace(opts, framework)
+    tag_name = opts[:tag_name] # This will be the string of the tag that we are using.
 
-    host_id = opts[:id]
-    tag_name = opts[:tag_name]
+    # Find the host entry in the current workspace by searching on
+    # the IP address we are currently processing, then save the database
+    # entry into the "host" variable.
+    # @type [Mdm::Host]
+    host = wspace.hosts.find_by_address(opts[:address])
```

Comment:
```suggestion
    if opts[:id]
       host = wspace.hosts.find(host_id)
    elsif opts[:address]
       host = wspace.hosts.find_by_address(opts[:address])
    else
       raise ...whatever the normal exception is...
    end
```

Reply:
Still verifying this but at the moment this seems to be causing the adding of tags to hosts to break with the current suggested change. I'll see if I can find out why this might be failing.

Reply:
Ah okay I think I remember why this wasn't working. The `id` here doesn't actually update at all like it should. This is why I never bothered to actually use it.

Reply:
Ah so apparently `host_data_proxy.rb` has a weird rule where its `add_host_id_to_opts` function will first check to see if the `opts[:id]` is set and if so then it will not update the ID. This leads to another bug whereby only the first ID is used and not later IDs.

Reply:
Should be fixed in https://github.com/rapid7/metasploit-framework/pull/17576/commits/2df445152766e46b99c1494fce5e67d87d022586 and the bug described above has also been addressed in this commit. 

@adfoster-r7 Could you let me know if this fix looks good or if you think additional steps are needed?